File: app/src/main/java/com/hinterlong/kevin/swishticker/query/Player.java
package com.hinterlong.kevin.swishticker.query;

public class Player {
    private String name;
    private int number;
    private int teamId;

    public Player(String name, int number) {
        this.name = name;
        this.number = number;
        this.teamId = -1;
    }

    public Player(String name, int number, int teamId) {
        this.name = name;
        this.number = number;
        this.teamId = teamId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public int getTeamId() {
        return teamId;
    }

    public void setTeamId(int teamId) {
        this.teamId = teamId;
    }
}

File: app/src/main/java/com/hinterlong/kevin/swishticker/service/data/Player.kt
package com.hinterlong.kevin.swishticker.service.data

import androidx.room.*

@Entity(
    tableName = "players",
    indices = arrayOf(
        Index("team_id")
    ),
    foreignKeys = arrayOf(
        ForeignKey(entity = Team::class,
            parentColumns = arrayOf("id"),
            childColumns = arrayOf("team_id"),
            onDelete = ForeignKey.CASCADE)
    )
)
data class Player(
    @ColumnInfo(name = "name") val name: String,
    @ColumnInfo(name = "team_id") val team: Long
) {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Long = 0
}